lib/git_worktree.rb

```diff
@@ -276,7 +278,8 @@ def fetch_and_merge
 
   def fetch
     with_credential_options do |cred_options|
-      @repo.fetch(@remote_name, cred_options)
+      options = cred_options.merge(:proxy_url => @proxy_url)
```

Comment:
Should this be part of the cred_options that come from with_credential_options?  Particularly because it will be required on every call, and it's sort of related to connections?

Reply:
Yeah, if we make it something more generic. Right now the proxy doesn't have anything to do with credentials (and could require its own credentials in the future) so the naming would get confusing.